src/unitxt/llm_as_judge.py

```diff
@@ -24,27 +32,105 @@ class LLMAsJudge(BulkInstanceMetric):
         template (Template): The template used when generating inputs for the judge llm.
         format (Format): The format used when generating inputs for judge llm.
         system_prompt (SystemPrompt): The system prompt used when generating inputs for judge llm.
-        strip_system_prompt_and_format_from_inputs (bool): Whether to strip the system prompt and formatting from the
-         inputs that the models that is being judges received, when they are inserted to the llm-as-judge prompt.
         inference_model (InferenceEngine): The module that creates the inference of the judge llm.
         reduction_map (dict): A dictionary specifying the reduction method for the metric.
         batch_size (int): The size of the bulk.
     """
 
     main_score: str = "llm_as_judge"
-    task: Literal[
-        "rating.single_turn",
-        "rating.single_turn_with_reference",
-        "pairwise_comparative_rating.single_turn",
-    ]
+    task: str
     template: Template
     system_prompt: SystemPrompt = Field(default_factory=EmptySystemPrompt)
     format: Format = Field(default_factory=SystemFormat)
-    strip_system_prompt_and_format_from_inputs: bool = True
     inference_model: InferenceEngine
     reduction_map: Optional[Dict[str, List[str]]] = None
     batch_size: int = 32
     prediction_type = Any  # Because handled with multiple tasks
+    infer_log_probs = False
+    include_meta_data: bool = False
```

Comment:
Why does the base class should be aware of these implementation parameters

Reply:
removed these from base class